docs/wip/exchange-onboarding.md

```diff
@@ -0,0 +1,261 @@
+
+# Wallet
+
+## Backup local state
+
+Skip to the next section if this is target machine doesn't yet have
+`cardano-sl` set up.
+
+To avoid catastrophic data loss, stop the wallet and backup the
+local databases, keys, certificates, and logs. By default, the
+local state will be in `./state-wallet-mainnet`, but may be
+elsewhere (see `stateDir` attribute in `./custom-wallet-config.nix`).
+
+## Fetch latest code
+
+Clone the [cardano-sl repository](https://github.com/input-output-hk/cardano-sl) or `cd` into a preexisting copy.
+
+    git clone https://github.com/input-output-hk/cardano-sl.git
+    cd cardano-sl
+
+Switch to the `master` branch and pull the latest code.
+
+    git checkout master
+    git pull
+
+Dump the current revision and confirm with IOHK whether it is as
+expected.
+
+    git rev-parse HEAD
+
+## Generate custom configuration
+
+The `cardano-sl` repo defaults to using end user topology and settings.
+
+Before building the wallet copy `./sample-wallet-config.nix` to
+`./custom-wallet-config.nix` and edit as needed.
+
+Supported options include:
+
+-   **`walletListen`:** Wallet API server
+-   **`ekgListen`:** Runtime metrics server
+-   **`stateDir`:** Directory for the wallet's local state. Must be
+    enclosed in double quotes.
+-   **`topologyFile`:** Used to connect to a custom set of nodes on
+    the network. When unspecified an appropriate
+    default topology is generated.
+
+For exchanges we recommend creating the following `custom-wallet-config.nix`:
+
+    # If any customization is required, copy this file to
+    # ./custom-wallet-config.nix and make edits there.
+    {
+      ## Wallet API server.
+      #walletListen = "127.0.0.1:8090";
+    
+      ## Runtime metrics server.
+      #ekgListen = "127.0.0.1:8000";
+    
+      ## Directory for the wallet's local state. Must be set BEFORE
+      ## running nix-build to have any effect, and it must be enclosed in
+      ## double quotes.
+      stateDir = "./state-wallet-mainnet";
+      topologyFile = ./exchange-topology.yaml;
+    
+      ## See https://downloads.haskell.org/~ghc/8.0.2/docs/html/users_guide/runtime_control.html#running-a-compiled-program
+      #ghcRuntimeArgs = "-N2 -qg -A1m -I0 -T";
+    
+      ## Primarily used for troubleshooting.
+      #additionalNodeArgs = "";
+    }
+
+
+The rest of this document will assume the above configuration file. Please
+alter any commands for example related to `stateDir` path to reflect your
+setup.
+
+You will also need to add the following `exchange-topology.yaml` file to use the
+private relays:
+
+    TODO: private relays should be inserted here in document given to exchanges
+
+## Building and running
+
+We support two different ways to build and run the wallet. You can run
+it standalone on any linux system built in the nix store or you can generate
+a docker container that can be ran on any docker container orchestrator like
+docker swarm or kubernetes.
+
+### Build and run in the nix store
```

Comment:
Could you prefix this heading with *Method 1:* and the next heading with *Method 2:*. This makes it clear to the skimmer that only one of the methods needs to be done.